Fig.Core/Settings.cs

```diff
@@ -154,45 +177,117 @@ public T Bind<T>(bool requireAll = true, string prefix = null) where T : new()
-                        .Invoke(this, new object[] { requireAll, name });
+                    if (preload && prop.Name == nameof(Environment)) continue;
+
+                    var readonlyProp = prop.GetSetMethod() is null;
+                    var type = prop.PropertyType;
+                    var name = String.IsNullOrEmpty(prefix?.Trim()) ? prop.Name : $"{prefix}.{prop.Name}";
+                    object value = null;
+
+                    if (preload || !readonlyProp)
+                    {
+                        if (!type.IsPrimitive
+                            && !type.IsEnum
+                            && type != typeof(string)
+                            && type != typeof(decimal)
+                            && type != typeof(DateTime)
+                            && type != typeof(TimeSpan))
+                        {
+                            var valObjectResult = this.GetType()
+                                .GetMethods(BindingFlags.NonPublic | BindingFlags.Instance)
+                                .Where(x => x.Name == "GetBindResult" && x.GetParameters()?.Length == 4)
+                                .FirstOrDefault()?
+                                .MakeGenericMethod(type)?
+                                .Invoke(this, new object[] { requireAll, name, false, errors });
+
+                            var typedResult = valObjectResult as IBindResult;
+                            if (typedResult != null)
+                            {
+                                if (typedResult.Errors?.Any() ?? false)
+                                {
+                                    errors.AddRange(typedResult.Errors);
+                                }
+
+                                value = typedResult.Result;
+                            }
+                        }
+                        else if (preload)
+                        {
+                            var getter = prop.GetGetMethod();
+                            if (getter == null) continue;
+
+                            value = getter.Invoke(this, Array.Empty<object>());
+                        }
+
+                        if (value is null && SettingsDictionary.TryGetValue(name, Environment, out var result) || requireAll)
+                        {
+                            value = Get(type, name);
+                        }
+
+                        if (!(value is null))
+                        {
+                            if (preload && !_cache.ContainsKey(name))
+                            {
+                                _cache[name] = new CacheEntry(value);
+                            }
+                            else if (!readonlyProp)
+                            {
+                                prop.SetValue(
+                                    target,
+                                    value
+                                );
+                            }
+
+                        }
+                    }
```

Comment:
This block handles nested types? Consider extracting to a method with an appropriate name

Reply:
I agree.  This would probably be better broken out.  Would definitely make it more readable.

Reply:
Broke this block out into a `GetPropertyValue` method and added some comments.